src/App.tsx

```diff
@@ -15,36 +15,81 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
 export const App: React.FC = () => {
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortType, setSortType] = useState(SortType.NONE);
+
+  function getReorderedGoods(goods: string[]) {
```

Comment:
Let's take it a step further and declare this function at a separate helpers file. This will require you to pass additional arguments into it, but will make your components structure cleaner.